src/menu_events.cpp

```diff
@@ -454,6 +454,39 @@ void menu_handler::show_enemy_moves(bool ignore_units, int side_num)
 			gui_->highlight_another_reach(path);
 		}
 	}
+	
+	//   Fetch cursor position, in order to later find a possible
+	// unit under it.
+	int x = -1;
+	int y = -1;
+	// auto bitmask =
+			SDL_GetMouseState(&x, &y);
+
+	//   Find possible unit (no matter whether friend or foe) under
+	// the cursor.
+	const mouse_handler& _mouse_handler =
+			const_cast<mouse_handler&>(
+					pc_.get_mouse_handler_base());
+	const game_display& _gd =
+			const_cast<game_display&>(
+					const_cast<mouse_handler&>(
+							_mouse_handler
+					).gui____public_for_demo____REMOVEME());
+	const map_location _hex_under_mouse = _gd.hex_clicked_on(x, y);
+	const auto _selected_hex_has_unit = _mouse_handler
+			.find_unit____public_for_demo____REMOVEME(
+					_hex_under_mouse
+			).valid();
+    
+	if (_selected_hex_has_unit) {
+		//   At this point, a single pixel move would remove the
+		// enemy [best] movements highlight hex tiles, so some
+		// prevention on normal unit mouseover movement highlight
+		// has to be toggled.
+
+		const_cast<mouse_handler&>(_mouse_handler)
+				.prevent_units_highlight();
```

Comment:
Is there any way we can eliminate the `const_cast` here? Declaring the `preventing_units_highlight_` variable mutable (and the functions const) would work, but I'm not sure if that's a good idea,.

Reply:
By what I've read, declaring the `preventing_units_highlight_` variable `mutable`, `prevent_units_highlight()` `const` and `enable_units_highlight()` `const` is equivalent to consider the variable part of the internal state instead of part of the externally visible state.

I think I don't know the codebase enough to provide a well informed answer and so the cautious way to do is to hold the `const_cast`. I would prefer leaving it like this.